File: problems/problem_003/golang/solution.go
package problem_003

import "math"

func bestTimeToBuyAndSell(prices []int) int {
	if len(prices) < 2 {
		return 0
	}

	minPrice := math.MaxFloat64
	maxProfit := 0

	for _, price := range prices {
		minPrice = math.Min(minPrice, float64(price))
		maxProfit = int(math.Max(float64(maxProfit), float64(price) - minPrice))
	}

	return maxProfit
}

File: problems/problem_003/javascript/solution.js
export function bestTimeToBuyAndSell(prices) {
    if (prices.length < 2) {
        return 0;
    }

    let [minPrice, maxProfit] = [Infinity, 0];

    for (let price of prices) {
        minPrice = Math.min(minPrice, price);
        maxProfit = Math.max(maxProfit, price - minPrice);
    }

    return maxProfit;
}